### PR TITLE
Remove getLocalTimeZoneOffset from Duktape.java.

### PR DESCRIPTION
**Pull request by szurbrigg**

### CODE REVIEW

File: duktape/src/main/jni/duktape-jni.cpp [hunk 1]
```diff
@@ -23,27 +24,31 @@
...
-  getLocalTimeZoneOffset = env->GetStaticMethodID(static_cast<jclass>(duktapeClass->get()),
-                                                  "getLocalTimeZoneOffset",
-                                                  "(D)I");
+
+  auto tzClass = env->FindClass("java/util/TimeZone");
+  auto getDefaultTimeZone = env->GetStaticMethodID(tzClass, "getDefault", "()Ljava/util/TimeZone;");
+  auto getOffset = env->GetMethodID(tzClass, "getOffset", "(J)I");
+
+  const GlobalRef timeZoneClass(env, tzClass);
+  getTimeZoneOffset = [timeZoneClass, getDefaultTimeZone, getOffset](double time) {
```
Comment (swankjesse) [inline duktape/src/main/jni/duktape-jni.cpp]: Explicitly declaring what's captured is interesting!

**Comment by szurbrigg** [comment]
Hmm, you think it'll make a big performance difference?  I just didn't like having that weird method hanging out in Java when I could completely hide it in the lib.

File: duktape/src/main/jni/duktape-jni.cpp (hunk 1 above)
Comment (szurbrigg) [reply duktape/src/main/jni/duktape-jni.cpp]: Safety first and no surprises.

**Comment by swankjesse** [comment]
Gotcha. No I definitely don't care about the performance cost here either way. It's completely negligible.

But I was trying to infer your motivation and the branch name _one-less-hop_ made me guess it was intended as an optimization. 

Keeping the Java code short & direct is great, and this helps.

LGTM!

**Pull request merged**